pkg/services/folder/folderimpl/unifiedstore.go

```diff
@@ -0,0 +1,467 @@
+package folderimpl
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/grafana/grafana/pkg/apimachinery/identity"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8sUser "k8s.io/apiserver/pkg/authentication/user"
+	k8sRequest "k8s.io/apiserver/pkg/endpoints/request"
+
+	"github.com/grafana/grafana/pkg/apis/folder/v0alpha1"
+	"github.com/grafana/grafana/pkg/infra/log"
+	internalfolders "github.com/grafana/grafana/pkg/registry/apis/folders"
+	"github.com/grafana/grafana/pkg/services/apiserver/endpoints/request"
+	"github.com/grafana/grafana/pkg/services/folder"
+	"github.com/grafana/grafana/pkg/setting"
+	"github.com/grafana/grafana/pkg/util"
+)
+
+type FolderUnifiedStoreImpl struct {
+	log       log.Logger
+	k8sclient folderK8sHandler
+}
+
+// sqlStore implements the store interface.
+var _ folder.Store = (*FolderUnifiedStoreImpl)(nil)
+
+func ProvideUnifiedStore(cfg *setting.Cfg) *FolderUnifiedStoreImpl {
+	k8sHandler := &foldk8sHandler{
+		gvr:        v0alpha1.FolderResourceInfo.GroupVersionResource(),
+		namespacer: request.GetNamespaceMapper(cfg),
+		cfg:        cfg,
+	}
+
+	return &FolderUnifiedStoreImpl{
+		k8sclient: k8sHandler,
+		log:       log.New("folder-store"),
+	}
+}
+
+func (ss *FolderUnifiedStoreImpl) Create(ctx context.Context, cmd folder.CreateFolderCommand) (*folder.Folder, error) {
+	newCtx, cancel, err := ss.getK8sContext(ctx)
+	if err != nil {
+		return nil, err
+	} else if cancel != nil {
+		defer cancel()
+	}
+
+	client, ok := ss.k8sclient.getClient(newCtx, cmd.OrgID)
+	if !ok {
+		return nil, nil
+	}
+
+	obj, err := internalfolders.LegacyCreateCommandToUnstructured(&cmd)
+	if err != nil {
+		return nil, err
+	}
+	out, err := client.Create(newCtx, obj, v1.CreateOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	folder, _ := internalfolders.UnstructuredToLegacyFolder(*out, cmd.SignedInUser.GetOrgID())
```

Comment:
```suggestion
	folder, err := internalfolders.UnstructuredToLegacyFolder(*out, cmd.SignedInUser.GetOrgID())
```

Reply:
The second parameter returned by `UnstructuredToLegacyFolder` is not an error, it's a string containing the `createdBy` user.

https://github.com/grafana/grafana/blob/0e953a8ff379364a5c29bb8b760a9985f4e99eee/pkg/registry/apis/folders/conversions.go#L67